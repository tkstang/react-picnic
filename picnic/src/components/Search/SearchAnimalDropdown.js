import React from 'react';
import { FormGroup, ControlLabel, FormControl } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { selectAnimal } from '../../actions';

const mapStateToProps = (state = {}, ownProps) => ({ SearchAnimalReducer: state.SearchAnimalReducer });

const mapDispatchToProps = dispatch => bindActionCreators({ selectAnimal }, dispatch);

const SearchAnimalDropdown = (props) => {
  const handleChange = (e) => {
    e.preventDefault();
    return props.selectAnimal(e.target.value);
  };

  return (
    <FormGroup>
      <FormControl
        componentClass="select"
        onChange={handleChange}
      >
        <option value="select">Select an Animal</option>
        <option value="cow">Cow</option>
        <option value="sheep">Sheep</option>
        <option value="goat">Goat</option>
        <option value="buffalo">Buffalo</option>
      </FormControl>
    </FormGroup>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchAnimalDropdown);
