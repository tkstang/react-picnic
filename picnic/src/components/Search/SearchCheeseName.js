import React from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { cheeseInput } from '../../actions';

const mapStateToProps = (state = {}, ownProps) => ({ CheeseNameReducer: state.CheeseNameReducer });

const mapDispatchToProps = dispatch => bindActionCreators({ cheeseInput }, dispatch);

const SearchCheeseName = (props) => {
  const handleChange = (e) => {
    e.preventDefault();
    console.log(e.target.value);
    return props.cheeseInput(e.target.value);
  };

  return (
    <FormGroup>
      <FormControl
        type="text"
        // value={this.state.value}
        placeholder="Enter a Cheese Name"
        onChange={handleChange}
      />
      <FormControl.Feedback />
    </FormGroup>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchCheeseName);
