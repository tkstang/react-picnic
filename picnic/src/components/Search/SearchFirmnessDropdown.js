import React from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { selectFirmness } from '../../actions';

const mapStateToProps = (state = {}, ownProps) => ({ SearchFirmnessReducer: state.SearchFirmnessReducer });

const mapDispatchToProps = dispatch => bindActionCreators({ selectFirmness }, dispatch);

const SearchFirmnessDropdown = (props) => {
  const handleChange = (e) => {
    e.preventDefault();
    return props.selectFirmness(e.target.value);
  };

  return (
    <FormGroup>
      <FormControl
        componentClass="select"
        onChange={handleChange}
      >
        <option value="select">Select a Firmness</option>
        <option value="hard">Hard</option>
        <option value="semi-hard">Semi-hard</option>
        <option value="semi-soft">Semi-soft</option>
        <option value="soft">Soft</option>
      </FormControl>
    </FormGroup>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchFirmnessDropdown);
