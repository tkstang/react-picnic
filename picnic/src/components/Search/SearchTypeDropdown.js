import React from 'react';
import { FormGroup, ControlLabel, FormControl } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { selectSearchType } from '../../actions';

const mapStateToProps = (state = {}, ownProps) => ({ SearchTypeReducer: state.SearchTypeReducer });

const mapDispatchToProps = dispatch => bindActionCreators({ selectSearchType }, dispatch);

const SearchTypeDropdown = (props) => {
  const handleChange = (e) => {
    e.preventDefault();
    return props.selectSearchType(e.target.value);
  };

  return (
    <FormGroup>
      <ControlLabel>How would you like to search for cheeses?</ControlLabel>
      <FormControl
        componentClass="select"
        placeholder="Select a Search Method"
        onChange={handleChange}
      >
        <option value="select">Select a Search Method</option>
        <option value="name">By Cheese Name</option>
        <option value="firmness">By Firmness</option>
        <option value="animal">By Animal</option>
      </FormControl>
    </FormGroup>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchTypeDropdown);

// class SearchTypeDropdown extends Component {
//   constructor(props) {
//     super(props);
//     this.handleChange = this.handleChange.bind(this)
//   }
//
//   handleChange(e) {
//     e.preventDefault();
//     return this.props.selectSearchType(e.target.value);
//   }
//
//   render() {
//     // console.log('this is selectSearchType', selectSearchType);
//     return (
//       <FormGroup controlId="typeSelect" ref={(searchType) => { this.input = searchType; }}>
//         <ControlLabel>How would you like to search for cheeses?</ControlLabel>
//         <FormControl
//           componentClass="select"
//           placeholder="Select a Search Method"
//           onChange={this.handleChange}
//         >
//           <option value="select">select</option>
//           <option value="name">By Cheese Name</option>
//           <option value="firmness">By Firmness</option>
//           <option value="animal">By Animal</option>
//         </FormControl>
//       </FormGroup>
//     );
//   }
// }

// SearchTypeDropdown.propTypes = {
//   selectSearchType: React.PropTypes.string.isRequired,
// };
