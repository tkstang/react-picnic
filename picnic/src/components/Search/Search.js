import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Grid, Row, Col, Button, Jumbotron } from 'react-bootstrap';
import renderIf from 'render-if';
import { Link } from 'react-router-dom';
import SearchTypeDropdown from './SearchTypeDropdown';
import SearchAnimalDropdown from './SearchAnimalDropdown';
import SearchFirmnessDropdown from './SearchFirmnessDropdown';
import SearchCheeseName from './SearchCheeseName';
import { getCheese } from '../../actions';


const mapDispatchToProps = dispatch => bindActionCreators({ getCheese }, dispatch);

const mapStateToProps = (state, ownProps) => {
  return {
    selection: state.searchType,
    animal: state.animal,
    firmness: state.firmness,
    name: state.name,
  };
};

export class Search extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    if (this.props.animal !== '') {
      this.props.getCheese(this.props.animal, this.props.selection);
    } else if (this.props.firmness !== '') {
      this.props.getCheese(this.props.firmness, this.props.selection);
    } else if (this.props.name !== ''){
      this.props.getCheese(this.props.name, this.props.selection);
    }
  }

  render() {
    const byAnimal = renderIf(this.props.selection === 'animal');
    const byFirmness = renderIf(this.props.selection === 'firmness');
    const byName = renderIf(this.props.selection === 'name');
    const isSearchReady = renderIf(
      (this.props.animal !== '') || (this.props.firmness !== '') || (this.props.name !== '')
    );

    return (
      <Grid>
        <Jumbotron>
          <h3>Welcome to React Picnic!</h3>
          <br />
          <p>React Picnic is the internet's premiere way to search for cheese!  Please enjoy your stay and search away!</p>
        </Jumbotron>
        <Row>
          <Col xs={12}>
            <SearchTypeDropdown />
            {byAnimal(<SearchAnimalDropdown />)}
            {byFirmness(<SearchFirmnessDropdown />)}
            {byName(<SearchCheeseName />)}
            {isSearchReady(<Link to="/results"><Button onClick={this.handleClick}>Search</Button></Link>)}
          </Col>
        </Row>
      </Grid>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);
// export default Search;
