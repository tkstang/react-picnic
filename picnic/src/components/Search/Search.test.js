import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import {Search} from './Search.js';

describe('Search', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
  });

  test('snapshots', () => {
    const component = shallow(
      <Search>Search</Search>
    );
    expect(toJson(component)).toMatchSnapshot();
    //
    // component.find('[name="toggle-preview"]').simulate('click');
    // expect(toJson(component)).toMatchSnapshot();
  });
});
