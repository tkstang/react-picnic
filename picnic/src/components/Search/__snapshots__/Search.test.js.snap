exports[`Search snapshots 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}>
  <Jumbotron
    bsClass="jumbotron"
    componentClass="div">
    <h3>
      Welcome to React Picnic!
    </h3>
    <br />
    <p>
      React Picnic is the internet\'s premiere way to search for cheese!  Please enjoy your stay and search away!
    </p>
  </Jumbotron>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <Connect(SearchTypeDropdown) />
      <Link
        replace={false}
        to="/results">
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}>
          Search
        </Button>
      </Link>
    </Col>
  </Row>
</Grid>
`;
